### PR TITLE
added network filtering to SSO

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Allows to filter which network is associated with a firewall SSO
# Impacts

Firewall SSO
# Delete branch after merge

YES
# NEWS file entries
## Enhancements
- Added ability to filter which IP networks are associated to a firewall for Single-Sign-On

### CODE REVIEW

**Comment by louismunro** [comment]
Please update description.

**Comment by julsemaan** [comment]
Description updated,

@louismunro please review

File: html/pfappserver/lib/pfappserver/Form/Config/Firewall_SSO.pm [hunk 1]
```diff
@@ -84,6 +84,13 @@ has_field 'cache_timeout' =>
    tags => { after_element => \&help,
              help => 'Adjust the "Cache timeout" to half the expiration delay in your firewall.<br/>Your DHCP renewal interval should match this value.' },
   );
+has_field 'networks' =>
+  (
+   type => 'Text',
+   label => 'Networks on which to do SSO',
+   tags => { after_element => \&help,
+             help => 'Comma delimited list of networks to apply the SSO on.<br/>Format : 192.168.0.0/24' },
```
Comment (louismunro) [inline html/pfappserver/lib/pfappserver/Form/Config/Firewall_SSO.pm]: Nitpick: "Comma delimited list of networks on which the SSO applies."

File: lib/pf/api.pm [hunk 2]
```diff
@@ -253,6 +253,10 @@ sub firewallsso : Public {
     foreach my $firewall_id ( sort keys %pf::config::ConfigFirewallSSO ) {
         my $firewall = pf::factory::firewallsso->new($firewall_id);
 
+        unless($firewall->should_sso($postdata{ip}, $postdata{mac})){
```
Comment (louismunro) [inline lib/pf/api.pm]: Nitpick: could be better written as "next unless".

**Comment by louismunro** [comment]
LGTM apart from minor nitpicks.
Fix them if you think it's worth it and then merge with my blessing.

**Comment by louismunro** [comment]
LGTM apart from minor nitpicks.
Fix them if you think it's worth it and then merge with my blessing.

**Pull request merged**